src/audio/asrc/asrc.c

```diff
@@ -268,94 +201,7 @@ static void src_copy_s16(struct processing_module *mod,
 	*n_written = out_frames;
 }
 
-#ifndef CONFIG_IPC_MAJOR_4
-static inline uint32_t asrc_get_source_rate(const struct ipc_config_asrc *ipc_asrc)
-{
-	return ipc_asrc->source_rate;
-}
 
-static inline uint32_t asrc_get_sink_rate(const struct ipc_config_asrc *ipc_asrc)
-{
-	return ipc_asrc->sink_rate;
-}
-
-static inline uint32_t asrc_get_operation_mode(const struct ipc_config_asrc *ipc_asrc)
-{
-	return ipc_asrc->operation_mode;
-}
-
-static inline bool asrc_get_asynchronous_mode(const struct ipc_config_asrc *ipc_asrc)
-{
-	return ipc_asrc->asynchronous_mode;
-}
-#else
-static inline uint32_t asrc_get_source_rate(const struct ipc4_asrc_module_cfg *ipc_asrc)
-{
-	return ipc_asrc->base.audio_fmt.sampling_frequency;
-}
-
-static inline uint32_t asrc_get_sink_rate(const struct ipc4_asrc_module_cfg *ipc_asrc)
-{
-	return ipc_asrc->out_freq;
-}
-
-static inline uint32_t asrc_get_operation_mode(const struct ipc4_asrc_module_cfg *ipc_asrc)
-{
-	return ipc_asrc->asrc_mode & (1 << IPC4_MOD_ASRC_PUSH_MODE) ? ASRC_OM_PUSH : ASRC_OM_PULL;
-}
-
-static inline bool asrc_get_asynchronous_mode(const struct ipc4_asrc_module_cfg *ipc_asrc)
-{
-	return false;
-}
-
-#endif
-
-static int asrc_init(struct processing_module *mod)
-{
-	struct comp_dev *dev = mod->dev;
-	struct module_data *mod_data = &mod->priv;
-#ifndef CONFIG_IPC_MAJOR_4
-	const struct ipc_config_asrc *ipc_asrc = mod_data->cfg.init_data;
-#else
-	const struct ipc4_asrc_module_cfg *ipc_asrc =  mod_data->cfg.init_data;
-#endif
```

Comment:
seems like an overkill to duplicate the whole function just because of one type mismatch. Maybe use a `typedef` or even just a `#define` in a header, e.g.
#ifdef CONFIG_IPC_MAJOR_4
typedef struct ipc4_asrc_module_cfg ipc_asrc_cfg;
#else
typedef struct ipc_config_asrc ipc_asrc_cfg;
#endif
```